Detectors/TRD/calibration/include/TRDCalibration/CalibratorGain.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file CalibratorGain.h
+/// \brief TimeSlot-based calibration of gain
+/// \author Ole Schmidt, Gauthier Legras
+
+#ifndef O2_TRD_CALIBRATORGAIN_H
+#define O2_TRD_CALIBRATORGAIN_H
+
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "DataFormatsTRD/Constants.h"
+#include "DataFormatsTRD/GainCalibHistos.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "DataFormatsTRD/CalGain.h"
+#include "TRDCalibration/CalibrationParams.h"
+
+#include "Rtypes.h"
+#include "TProfile.h"
+#include "Fit/Fitter.h"
+#include "TFile.h"
+#include "TTree.h"
+
+#include <array>
+#include <cstdlib>
+#include <memory>
+
+namespace o2
+{
+namespace trd
+{
+
+class CalibratorGain final : public o2::calibration::TimeSlotCalibration<o2::trd::GainCalibHistos>
+{
+  using Slot = o2::calibration::TimeSlot<o2::trd::GainCalibHistos>;
+
+ public:
+  CalibratorGain(bool enableOut = false) : mEnableOutput(enableOut) {}
+  ~CalibratorGain() final = default;
+
+  bool hasEnoughData(const Slot& slot) const final { return slot.getContainer()->getNEntries() >= mMinEntriesTotal; }
+  void initOutput() final;
+  void finalizeSlot(Slot& slot) final;
+  Slot& emplaceNewSlot(bool front, TFType tStart, TFType tEnd) final;
+
+  /// (Re-)Creates a file "trd_calibgain.root". This lets continually fill
+  /// a tree with the fit results and the residuals histograms.
+  void createOutputFile();
+
+  /// Close the output file. E.g. First write the tree to the file and let the
+  /// smart pointers take care of closing the file.
+  void closeOutputFile();
+
+  const std::vector<o2::trd::CalGain>& getCcdbObjectVector() const { return mObjectVector; }
+  std::vector<o2::ccdb::CcdbObjectInfo>& getCcdbObjectInfoVector() { return mInfoVector; }
+
+  void initProcessing();
+
+  /// Initialize the fit values once with the previous valid ones if they are
+  /// available.
+  void retrievePrev(o2::framework::ProcessingContext& pc);
```

Comment:
You are not doing that. Why define at all, just initialize with you default value.